### ISSUE TITLE
Blade Pro 2015

### ISSUE THREAD

**Issue by 3cpo** [openrazer/openrazer#360]
Hi,

Is there any track pad support for Blade Pro 2015?  I've been using:
 * Razer USB keyboard device driver
 *  Copyright (c) 2014 Milan Votava (votava@mageo . cz)
 *  Copyright (c) 2013 Zach Carlson (FxChiP@Gmail . com)
 * Bases on:	usb_skeleton.c v2.2 by Greg Kroah-Hartman
Above does not support track pad graphics.  Is there anything available that supports track pad graphics and added buttons?
Many Thanks!

**Comment by terricain** [comment]
Nothing supports it yet not this project

**Comment by z3ntu** [comment]
The output of lsusb please

**Comment by 3cpo** [comment]
Here you go:

lsusb
Bus 002 Device 002: ID 8087:8000 Intel Corp. 
Bus 002 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 001 Device 002: ID 8087:8008 Intel Corp. 
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 004 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 003 Device 004: ID 8087:07dc Intel Corp. 
Bus 003 Device 015: ID 1532:0116 Razer USA, Ltd 
Bus 003 Device 002: ID 0bda:57a2 Realtek Semiconductor Corp.                                                                     
Bus 003 Device 016: ID 25a7:fa23                                                                                                 
Bus 003 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub    

I have a rule to load the driver that was referenced in my earlier post:
ACTION=="add", DRIVER=="usbhid", ATTRS{idVendor}=="1532", ATTRS{idProduct}=="0116", RUN+="/usr/local/razer-0.01/razer-bind-udev.sh

**Comment by terricain** [comment]
Can you get me some screenshots from synapse showing what the trackpad can do

**Comment by 3cpo** [comment]
Hi terrycain,

Sorry for the delay.  Here are some screenshots.  Thanks for looking at this!
![synapticsfigure1](https://user-images.githubusercontent.com/31116998/30290566-091451ba-96ed-11e7-84bd-b79341a6b5b2.png)
![synapticsfigure5](https://user-images.githubusercontent.com/31116998/30290563-090fe3a0-96ed-11e7-892f-7b6677f4097d.png)
![synapticsfigure2](https://user-images.githubusercontent.com/31116998/30290565-0912deac-96ed-11e7-920b-61f7a8febfbf.png)
![synapticsfigure3](https://user-images.githubusercontent.com/31116998/30290567-091ae8fe-96ed-11e7-9da8-b77e96f8aef4.png)
![synapticsfigure4](https://user-images.githubusercontent.com/31116998/30290564-091178e6-96ed-11e7-879b-e51273ea0ff1.png)

**Comment by z3ntu** [comment]
@3cpo We need screenshots from the Windows Application called "Razer Synapse" as that'll show us what hardware we are dealing with. The "Prerequisites" section on [this wiki page](https://github.com/openrazer/openrazer/wiki/Reverse-Engineering-USB-Protocol) will tell you more.

**Comment by zackteo** [comment]
Is there still no support for Blade Pro 2015?

**Comment by z3ntu** [comment]
@zackteo No, as we currently don't have the necessary information about the device that we need.

* Read [this](https://github.com/openrazer/openrazer/wiki/Reverse-Engineering-USB-Protocol) because we need packet captures.
* Screenshots of all available options on Razer Synapse are needed.
* And lastly, the product & model number of the device for documentation reasons please.

**Comment by zackteo** [comment]
@z3ntu Hello, thanks for your super prompt response! Here are the screenshots from Razer Synapse :)
The last image is the window that appears when you click on apps. Synapse was bugging out (and the apps button didn't do anything) so I got that image from https://www.techspot.com/review/995-razer-blade-pro-2015/page2.html

![razerbladepro1](https://user-images.githubusercontent.com/7510179/41204309-611d711c-6d15-11e8-9e2b-9d8b28c03d0d.PNG)
![razerbladepro1 1](https://user-images.githubusercontent.com/7510179/41204312-6910791e-6d15-11e8-961a-84bdd5402237.PNG)
![razerbladepro1 2](https://user-images.githubusercontent.com/7510179/41204314-6c1b3630-6d15-11e8-8de9-6279363d06a7.PNG)
![razerbladepro2](https://user-images.githubusercontent.com/7510179/41204318-76c9bd40-6d15-11e8-98b1-f46ce675b96a.PNG)
![razerbladepro3](https://user-images.githubusercontent.com/7510179/41204319-76fbfd64-6d15-11e8-9416-6b4f6dea757f.PNG)
![razerbladepro4](https://user-images.githubusercontent.com/7510179/41204320-772ac414-6d15-11e8-95f2-748677477529.PNG)
![razerbladepro5](https://user-images.githubusercontent.com/7510179/41204435-04de820e-6d17-11e8-9321-c265788084ea.PNG)
![2015-04-28-image-10](https://user-images.githubusercontent.com/7510179/41204355-09cf5870-6d16-11e8-98d3-05fbcf9d8f15.jpg)

Do I also need to download wireshark on linux and proceed with phase 2? [Based on this](https://github.com/openrazer/openrazer/wiki/Reverse-Engineering-USB-Protocol)

**Comment by z3ntu** [comment]
The idea is having a Linux host with Wireshark capturing the usb traffic going in and out of the virtual machine where the usb device is passed through. If you just have Windows outside of a VM, you can also try capturing it directly in Windows, but I don't know if this'll work.

**Comment by zackteo** [comment]
In that case, since I am dual booting Ubuntu and windows 10. I just download wireshark on windows 10 and run it there ?

**Comment by z3ntu** [comment]
That could work, yes.
